src/utils/QueryParamRouteAdapter.tsx

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useMemo } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+
+/**
+ * Work around until fixed.
+ *
+ * Https://github.com/pbeshai/use-query-params/issues/108.
+ *
+ * This is the main thing you need to use to adapt the react-router v6
+ * API to what use-query-params expects.
+ *
+ * Pass this as the `ReactRouterRoute` prop to QueryParamProvider.
+ */
+export const QueryParamRouteAdapter = ({ children }: any) => {
```

Comment:
```suggestion
export const queryParamRouteAdapter = ({ children }: any) => {
```

Reply:
I think we should stick to PascalCase for components

Reply:
Also, I don't want to spend a lot of time, trying to properly typing it. I am even not sure if it can be properly typed.
It is working, and its bad typings do not affect anything.